doc/user-guide/oauthn_token_exchange.md

```diff
@@ -0,0 +1,124 @@
+## OAuthN Token exchange support
+
+The current release of Indigo IAM implements an essential set of feature of the
+[Token Exchange OAuth specification](https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-05).
+Indigo IAM supports only impersonation semantic: delegation is not yet supported.
+
+### Configuration
+Clients that request Token Exchange must be configured with
+the `urn:ietf:params:oauth:grant-type:token-exchange` grant type enabled.
+
+### Usage
+A client who wants to exchange an access token with a new one, must send a request to the `/token` endpoint,
+with the properties described below:
+
+1. Supply its valid access token in `subject_token` request field.
+2. Specify a valid client_id in `audience` field: this is the ID of the target resource which client wants to access.
+3. Requested scopes must be a subset of the scopes enabled on the client specified as `audience`.
+
+There are some scopes that are considered "specials".
+They are identity, refresh and SCIM related scopes:
+
+`openid`, `profile`, `email`, `address`, `phone`, `offline_access`, `scim:read`, `scim:write`
+
+These scopes, in order to be "exchanged" across services, need to be present in the set of
+scopes linked to the subject token that is presented for the exchange.
+
+
+### Limitation and known issues
+The current implementation of Token Exchange in Indigo IAM, has some limitations.
+
+ * Delegation is not yet supported: if `actor_token` or the flag `want_composite` are specified within the request, an error
+ response is returned by the authorization server.
+ * The `audience` field is mandatory and not optional, as mentioned into the specification: it must be a valid client identifier.
+ * The `resource` field is ignored.
+
+ 
+### Token Exchange example
+This section describe a typical scenario of token exchange flow.
+
+A client requests to the Authorization Service (AS) an access token (T1) to access a protected resource (PR).
+In this example, we use a registered client with `client_credentials` grant type to obtain the token and `curl` to send the HTTP request.
+
+```bash
+ $ export IAM_TOKEN_ENDPOINT=https://iam.local.io/token
+ $ export CLIENT_ID=token-exchange-subject
+ $ export CLIENT_SECRET=secret
+
+ $ curl -sH "Content-Type: application/x-www-form-urlencoded" \
+    --data "client_id=$CLIENT_ID&client_secret=$CLIENT_SECRET&grant_type=client_credentials&scope=openid%20profile" \
```

Comment:
use multiple -d instead. no need to specify the content-type. like in the following example:
```
curl -L \ 
   -d client_id=062e89c9-4bc8-4019-985f-38cc5cac81a0 \ 
   -d client_secret=${CLIENT_SECRET} \
   -d grant_type=password \
   -d scope="openid profile" ...
```